### DISCUSSION TITLE
taro2.2.17版本小程序首页列表渲染不丝滑问题

### DISCUSSION THREAD

**Discussion by pino-wd** [NervJS/taro#10827] [Q & A]
我司项目是一款基于taro2.2.17的微信电商小程序，首页主体就是一个商品列表，每个商品图与屏同宽。当分页获取商品数目超过一定数目时（大概30个），会明显感觉列表继续分页获取商品与商品图渲染卡顿，甚至某些机型下拉重新更新列表会卡3s以上（hm系统~）。
当下已经对首页的图片进行过适度的压缩处理了，但是没有明显改善。另一个板块的列表的商品浏览就很正常，唯一的大一些的区别可能就是页面布局，首页列表每项占一屏宽，此页是半屏宽，所以图片显示尺寸大概是4倍的差别。请教一下各位大佬，问题可能在哪些方面？有哪些做法可以优化呢？

**Comment by Chen-jj** [comment]
1. 打印小程序的 `setData` 调用，看看性能瓶颈在哪里。
2. 减少使用组件，尽量使用 render function 代替，如：

```jsx
// before
class List extends Components {
  render () {
    return (
      {this.list.map(() => <Item>)}
    )
  }
}
// after
class List extends Components {
  renderItem () {
    return <div />
  }
  render () {
    return (
      {this.list.map(() => renderItem())}
    )
  }
}
```
